File: packages/frontend/src/types/pet.ts
// 宠物实体类型定义
export interface Pet {
  id: number;
  name: string;           // 宠物姓名
  studentId: string;      // 学号 (例如 "P20240001")
  species: string;        // 种类（狗、猫等）
  breed?: string;         // 品种
  age: number;            // 年龄（月）
  weight: number;         // 体重（kg）
  gender: number;         // 1-公，2-母
  genderLabel?: string;   // 性别标签（公/母）
  photoUrl: string;       // 可爱照片 URL
  description?: string;   // 简介
  // 详细信息字段（详情页展示）
  ownerName?: string;
  ownerContact?: string;
  vaccinationDate?: string;
  isVaccinated?: number;  // 0-否，1-是
  isNeutered?: number;    // 0-否，1-是
  healthStatus?: string;
  status?: number;
  createTime?: string;
  updateTime?: string;
}

// 轮播图图片类型
export interface CarouselImage {
  id: number;
  imageUrl: string;
  title?: string;
  linkUrl?: string;
  sortOrder?: number;
  status?: number;
  createTime?: string;
}

// API 响应类型
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
}

// 分页响应类型
export interface PaginatedResponse<T> {
  list: T[];
  total: number;
  pages: number;
  pageNum: number;
  pageSize: number;
}

// 宠物登记表单数据
export interface PetRegisterFormData {
  name: string;           // 宠物姓名
  studentId?: string;     // 学号（可选，系统可自动生成）
  species: string;        // 种类（狗/猫/其他）
  breed?: string;         // 品种
  age: number;            // 年龄
  ageUnit: 'month' | 'year'; // 年龄单位
  weight: number;         // 体重（kg）
  gender: 'male' | 'female';
  photoUrl?: string;      // 照片 URL（可选）
  photoFile?: File;       // 照片文件（用于上传）
  description?: string;   // 简介
  ownerName: string;      // 主人姓名
  ownerContact: string;   // 主人联系方式
  isVaccinated: boolean;  // 是否已接种疫苗
  vaccinationDate?: string; // 疫苗接种日期
  isNeutered: boolean;    // 是否已绝育
  healthStatus?: string;  // 健康状况说明
}

File: packages/backend/src/main/java/com/petschool/entity/Pet.java
package com.petschool.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 宠物信息实体类
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@TableName("pet")
public class Pet {

    /**
     * 主键 ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 创建用户ID
     */
    private Long userId;

    /**
     * 学号（格式：P20240001）
     */
    private String studentId;

    /**
     * 宠物姓名
     */
    private String name;

    /**
     * 种类（狗、猫等）
     */
    private String species;

    /**
     * 品种
     */
    private String breed;

    /**
     * 年龄（月）
     */
    private Integer age;

    /**
     * 体重（kg）
     */
    private BigDecimal weight;

    /**
     * 性别：1-公，2-母
     */
    private Integer gender;

    /**
     * 照片 URL
     */
    private String photoUrl;

    /**
     * 简介
     */
    private String description;

    /**
     * 主人姓名
     */
    private String ownerName;

    /**
     * 主人联系方式
     */
    private String ownerContact;

    /**
     * 最近疫苗接种日期
     */
    private LocalDate vaccinationDate;

    /**
     * 是否已接种疫苗：0-否，1-是
     */
    private Integer isVaccinated;

    /**
     * 是否已绝育：0-否，1-是
     */
    private Integer isNeutered;

    /**
     * 健康状态备注
     */
    private String healthStatus;

    /**
     * 状态：0-禁用，1-正常
     */
    private Integer status;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;
}
